src/droplet.F

```diff
@@ -8,13 +8,20 @@ MODULE droplet_module
   implicit none
 
   private
-  public :: droplet_driver,droplet_diag
+  public :: droplet_driver,droplet_diag,initialize_droplet_histograms
 
 
 
   !These are quantities which are needed in the droplet_diag output, so must be public
   integer :: num100,num1000
-  real :: u10_ssgf,num_destroyed,num_injected
+  integer, parameter :: nbins = 514  !Bins for the droplet histograms
+  real,dimension(nbins) :: bins_restime,hist_restime  !Histograms for droplet residence time
+  real,dimension(nbins) :: bins_radius,hist_radius,hist_radius10,hist_fallrad  !Histograms for droplet radius
+  real :: restime_min,restime_max,restime_min10,restime_max10,dhres !Bin parameters for droplet residence time
+  real :: dhrad,rmin,rmax,rmin10,rmax10  !Bin parameters for droplet size
+
+  real :: u10_ssgf 
+  real :: num_destroyed,num_injected,mass_destroyed,mass_injected !Cumulative particle counts between droplet_diag calls
```

Comment:
@drichte2 Do these variables have to be module variables? They are used in the `acc routine` and involved in the reduction operation, which makes the GPU implementation difficult. Would like to see @johnmauff's option, too.

Reply:
Technically, there would be a way to remove these as module variables, but it'd be a pain. They'd have to be sent out of the droplet_driver and then into droplet_diag, which would mean declaring them somewhere at the cm1.F level. I suppose this could be done, but this seemed like a more compact way of doing it. 

For what it's worth, @johnmauff said in an email when I asked about this, "I don't think this should cause a problem for the GPU port, however, I want to hear Jian's opinion as well.   We can just mark the new data structures as device resident and make sure that the correct version gets updated."

Reply:
Generally I think John's comment is correct. However, in this particular case, those module variables are involved in a reduction operation in an acc routine, and I feel it is not trivial to get the code right on the GPU. I have added the necessary GPU directives (I think) and work on the testing now.

Reply:
I am not particularly supportive of adding additional arguments that are passed through the subroutine argument list.  However, Jian is correct that the use of module variables complicates the GPU implementation.  Currently, CM1 has a qbudget array that stores a number of different global sums related to moisture.  Maybe something similar could be done with droplet statistics if Jian's testing on GPU proves problematic.